src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -420,14 +437,36 @@ private void WatchForFileSystemEventsThreadStart(object arg)
                             }
                         }
                     }
+
+                    ArraySegment<char> underlyingArray;
+                    if (MemoryMarshal.TryGetArray(events[i], out underlyingArray))
+                        ArrayPool<char>.Shared.Return(underlyingArray.Array);
+                }
+
+                unsafe void ProcessEvents()
+                {
+                    for (int i = 0; i < events.Length; i++)
+                    {
+                        int endPoint = GetIndex(eventPaths[i]);
+                        Debug.Assert(endPoint > 0, "Empty events are not supported");
+                        events[i] = new Memory<char>(ArrayPool<char>.Shared.Rent(endPoint));
+                        int charCount;
+
+                        // Converting an array of bytes to UTF-8 char array
+                        fixed(char* temp = &MemoryMarshal.GetReference(events[i].Span))
```

Comment:
Can you use the [span overload](https://apisof.net/catalog/System.Text.Encoding.GetChars(ReadOnlySpan%3CByte%3E,Span%3CChar%3E))?